app/code/Magento/InventoryInStorePickup/Model/NotifyOrderIsReadyAndShip.php

```diff
@@ -0,0 +1,80 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\InventoryInStorePickup\Model;
+
+use Magento\InventoryInStorePickupApi\Api\IsOrderReadyForPickupInterface;
+use Magento\InventoryInStorePickupApi\Api\NotifyOrderIsReadyAndShipInterface;
+use Magento\InventoryInStorePickupApi\Exception\OrderIsNotReadyForPickupException;
+
+class NotifyOrderIsReadyAndShip implements NotifyOrderIsReadyAndShipInterface
+{
+    /**
+     * @var \Magento\InventoryInStorePickupApi\Api\IsOrderReadyForPickupInterface
+     */
+    private $isOrderReadyForPickup;
+
+    /**
+     * @var \Magento\Sales\Api\ShipOrderInterface
+     */
+    private $shipOrder;
+
+    /**
+     * @var \Magento\InventoryInStorePickup\Model\Order\Email\ReadyForPickupNotifier
+     */
+    private $emailNotifier;
+
+    /**
+     * @var \Magento\Sales\Api\OrderRepositoryInterface
+     */
+    private $orderRepository;
+
+    /**
+     * NotifyOrderIsReadyAndShip constructor.
+     *
+     * @param \Magento\InventoryInStorePickupApi\Api\IsOrderReadyForPickupInterface    $isOrderReadyForPickup
+     * @param \Magento\Sales\Api\ShipOrderInterface                                    $shipOrder
+     * @param \Magento\InventoryInStorePickup\Model\Order\Email\ReadyForPickupNotifier $emailNotifier
+     * @param \Magento\Sales\Api\OrderRepositoryInterface                              $orderRepository
+     */
+    public function __construct(
+        IsOrderReadyForPickupInterface $isOrderReadyForPickup,
+        \Magento\Sales\Api\ShipOrderInterface $shipOrder,
+        Order\Email\ReadyForPickupNotifier $emailNotifier,
+        \Magento\Sales\Api\OrderRepositoryInterface $orderRepository
+    ) {
+        $this->isOrderReadyForPickup = $isOrderReadyForPickup;
+        $this->shipOrder = $shipOrder;
+        $this->emailNotifier = $emailNotifier;
+        $this->orderRepository = $orderRepository;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function execute(int $orderId):?int
```

Comment:
Should recheck if we can replace it with `int`

Reply:
why not use `void`? That's a typical command

Reply:
I checked how this is implemented for the pure order shipping: `\Magento\Sales\Api\ShipOrderInterface::execute`

Reply:
Initially, I checked how the similar command was declared in `\Magento\Sales\Api\ShipOrderInterface::execute`.

However, as already discussed, we have not a 'real' shipment there, so I have fixed that.